### PR TITLE
Updating minimum PHP version to 5.6

### PR DESCRIPTION
**Pull request by rawkode**
I am targetting this branch, because it's a PHP version change / potential BC break

Refs # https://github.com/sonata-project/dev-kit/issues/208

## Changelog

``` markdown
### Deprecated
- PHP versions 5.3 - 5.5
```

## Subject

Updating minimum PHP version to 5.6, as per https://github.com/sonata-project/dev-kit/issues/208

### CODE REVIEW

**Comment by greg0ire** [comment]
Rebasing on the latest 3.x should fix StyleCI

**Comment by greg0ire** [comment]
ping @rawkode

**Comment by rawkode** [comment]
Looks like `buildTwigLikeUrl` needs removed from `SonataAdminExtensionTest`, but from the comments I don't understand why it exists.

Should it be replaced with `htmlspecialchars(http_build_query($url, '', '&'))` ?

**Comment by greg0ire** [comment]
What makes you say that and why don't you just rebase so that styleCI passes?

**Comment by rawkode** [comment]
Rebased. My previous comment was based on:

```
   9     /**
   8      * This method generates url part for Twig layout. Allows to keep BC for PHP 5.3.
   7      *
   6      * Remove this method for next major release only if PHP 5.3 support will be dropped.
   5      *
   4      * @param array $url
   3      *
   2      * @return string
   1      */
2107     private function buildTwigLikeUrl($url)  
```

**Comment by greg0ire** [comment]
Oh yes I remember that. See #3843

**Comment by greg0ire** [comment]
I think you can use `url_encode` directly now.

**Comment by rawkode** [comment]
@core23 Can you advise on this failure? I've added the `throw` annotation to the interface

**Comment by greg0ire** [comment]
1. Go to [this page](https://styleci.io/analyses/XNBRQy)
2. Download the patch
3. Apply the with `git apply /path/to/the/patch.txt`
4. commit and push

**Comment by rawkode** [comment]
Thanks for your help and patience, @greg0ire. I've applied the patch, but still getting the error with regards to an unexpected `throw`. This annotation was added to the `AdminInterface` on `setSujbect`, and `setSubject` on the concrete has `{inheritDoc}`, but that doesn't seem to help.

File: Twig/GlobalVariables.php [hunk 1]
```diff
@@ -20,39 +20,23 @@
...
-                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since 
-                version 3.5 and will be removed in 4.0. Use Sonata\AdminBundle\Admin\Pool instead.',
-                E_USER_DEPRECATED
-            );
-
-            $this->adminPool = $adminPool->get('sonata.admin.pool');
-        } elseif ($adminPool instanceof Pool) {
-            $this->adminPool = $adminPool;
-        } else {
+        if (false === ($adminPool instanceof Pool) {
```
Comment (greg0ire) [inline Twig/GlobalVariables.php]: You are missing a closing parenthesis here :wink:

Comment (greg0ire) [inline Twig/GlobalVariables.php]: You could just write if `(! $adminPool instanceof Pool)`, BTW

Comment (rawkode) [inline Twig/GlobalVariables.php]: Silly me! :sweat_smile: 

Thanks

File: Admin/AbstractAdmin.php [hunk 2]
```diff
@@ -577,14 +565,6 @@ public function getDataSourceIterator()
...
             $label = $this->getTranslationLabel($field, 'export', 'label');
             $transLabel = $this->trans($label);
-
-            // NEXT_MAJOR: Remove this hack, because all field labels will be translated with the major release
-            // No translation key exists
-            if ($transLabel == $label) {
-                $fields[$key] = $field;
-            } else {
-                $fields[$transLabel] = $field;
-            }
```
Comment (greg0ire) [inline Admin/AbstractAdmin.php]: This does not look right, `$transLabel` is unused. I think line 589 should be kept.

Comment (greg0ire) [inline Admin/AbstractAdmin.php]: This does not look right, $transLabel is unused. I think line 589 should be kept.

File: Admin/AdminInterface.php [hunk 3]
```diff
@@ -201,11 +185,9 @@ public function getModelManager();
     public function getManagerType();
 
     /**
-     * @param string $context NEXT_MAJOR: remove this argument
-     *
      * @return ProxyQueryInterface
      */
-    public function createQuery($context = 'list');
+    public function createQuery();
```
Comment (greg0ire) [inline Admin/AdminInterface.php]: I think similar methods in `AdminExtension` and `AdminExtensionInterface` have also a `$context` argument, which should also be removed.

File: Twig/GlobalVariables.php [hunk 4]
```diff
@@ -20,39 +20,24 @@
...
-                'Using an instance of Symfony\Component\DependencyInjection\ContainerInterface is deprecated since 
-                version 3.5 and will be removed in 4.0. Use Sonata\AdminBundle\Admin\Pool instead.',
-                E_USER_DEPRECATED
-            );
-
-            $this->adminPool = $adminPool->get('sonata.admin.pool');
-        } elseif ($adminPool instanceof Pool) {
-            $this->adminPool = $adminPool;
-        } else {
+        if (!$adminPool instanceof Pool) {
```
Comment (greg0ire) [inline Twig/GlobalVariables.php]: Please use type hinting instead.

**Comment by OskarStark** [comment]
@rawkode could you apply the comments of @greg0ire or shall we close this PR?

**Pull request closed without merging**